OpenSim/Simulation/SimbodyEngine/BallJoint.h

```diff
@@ -52,14 +52,36 @@ class OSIMSIMULATION_API BallJoint : public Joint {
     OpenSim_DECLARE_CONCRETE_OBJECT(BallJoint, Joint);
 
     /** Specify the Coordinates of the BallJoint */
-    CoordinateIndex rx{ constructCoordinate(Coordinate::MotionType::Rotational) };
-    CoordinateIndex ry{ constructCoordinate(Coordinate::MotionType::Rotational) };
-    CoordinateIndex rz{ constructCoordinate(Coordinate::MotionType::Rotational) };
+    CoordinateIndex rx{constructCoordinate(Coordinate::MotionType::Rotational)};
+    CoordinateIndex ry{constructCoordinate(Coordinate::MotionType::Rotational)};
+    CoordinateIndex rz{constructCoordinate(Coordinate::MotionType::Rotational)};
 
 public:
     /** Use Joint's constructors. @see Joint */
     using Joint::Joint;
 
+    using Joint::getCoordinate;
+
+    /** Indices of Coordinates; can be used with getCoordinate() for
+        convenience. */
+    enum class Coord {
+        Rotation1,
+        Rotation2,
+        Rotation3
+    };
+
+    /** Get the Coordinates associated with this Joint. */
+    const Coordinate& getCoordinate(Coord idx) const {
```

Comment:
Thanks for explaining. Silly me...I just finally saw that there *is* a method in the base class.

> For example, a user could change the joint type without changing the argument to getCoordinate() and end up with the wrong index into the CoordinateSet.

That is a nice feature of the current implementation.

> including all the enums in Joint seems undesirable.

I agree; this would not be a good idea.


> most elegant to have a single getCoordinate() that was common to all joint types.

[This post](http://stackoverflow.com/questions/411103/function-with-same-name-but-different-signature-in-derived-class), which is about the `using Joint::getCoordinate` line, disagrees: 

> It is generally considered bad practice to have have functions in derived classes which have the same name as functions in the bass class which aren't intended to override the base class functions as what you are seeing is not usually desirable behaviour. It is usually preferable to give different functions different names.

I can understand that sentiment. It's not so bad in this case because the methods really are doing pretty much the same thing. If we leave in that `using` line, we should perhaps explain why it's necessary.


Anyway, I am fine with this PR.